lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp

```diff
@@ -93,6 +117,8 @@ void addElementwiseUnaryOpsConversionPatterns(MLIRContext *ctx,
   patterns.add<TosaToTTIRDefaultDPSOpConversionPattern<tosa::CeilOp,
                                                        mlir::tt::ttir::CeilOp>>(
       typeConverter, ctx);
+  patterns.add<TosaToTTIRClampOpConversionPattern>(typeConverter, ctx);
+
```

Comment:
Move this to the bottom, keep `TosaToTTIRDefaultDPSOpConversionPattern` grouped. 

Blank line unnecessary.

Reply:
Moved, along with pattern for multiply op that also doesn't (directly) use the `TosaToTTIRDefaultDPSOpConversionPattern`. 
Also moved corresponding tests to `have_special_pattern` folder.

Reply:
Yeah, we should keep organization similar to `lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp`.

Not so sure about the name `have_special_pattern` though. This will grow over time and a bunch of unrelated ops will end up there. @mrakitaTT wdy think? Stablehlo folder is not relevant here since it is messy.

Reply:
Sorry for late response. I think we should keep these organised by type of ops, not by type of patterns used. We should also group pattern adds for same type of ops into functions, but if some op is specific and doesn't fall into some general type of ops then we should add its pattern directly inside `populateTosaToTTIRPatterns` function. You can add more patterns with a single `.add` call like this:
```
patterns.add<TosaToTTIRClampOpConversionPattern,
             SomeOtherSpecificOpConversionPattern,
             SomeThirdSpecificOpConversionPattern>(typeConverter, ctx);
```

So for example having function like `addElementwiseBinaryOpsConversionPatterns ` is fine, it groups pattern adds for all eltwise binary ops, and we should return `TosaToTTIRMultiplyOpConversionPattern` there. On the other hand there is not much point having function `addClampOpConversionPattern` which adds pattern for just one op.

Here is how that would look like:
```
void populateTosaToTTIRPatterns(MLIRContext *ctx, RewritePatternSet &patterns,
                                TypeConverter &typeConverter) {
  addElementwiseUnaryOpsConversionPatterns(ctx, patterns, typeConverter);
  addElementwiseBinaryOpsConversionPatterns(ctx, patterns, typeConverter);
  addElementwiseTernaryOpsConversionPatterns(ctx, patterns, typeConverter);
  addLogicalOpsConversionPatterns(ctx, patterns, typeConverter);
  addCompareOpsConversionPatterns(ctx, patterns, typeConverter);

  patterns.add<TosaToTTIRClampOpConversionPattern,
               SomeOtherSpecificOpConversionPattern,
               SomeThirdSpecificOpConversionPattern>(typeConverter, ctx);
}
```

Reply:
Regarding test organization, let's do the same. You should create folders inside `test/ttmlir/Conversion/TOSAToTTIR/` for the type of ops that you have grouped into separate functions from `populateTosaToTTIRPatterns`. So for example you should have `test/ttmlir/Conversion/TOSAToTTIR/elementwise_unary/`, `test/ttmlir/Conversion/TOSAToTTIR/elementwise_binary/`, etc. And tests for other ops like `Clamp` should be added directly in root folder `test/ttmlir/Conversion/TOSAToTTIR/`. Each op should have one file testing it, with multiple test cases inside if needed (each test case is another mlir func inside the module). You can also have test files which do not test specific op, but for example data type conversion, feel free to add them to root folder too.

Reply:
FYI @sgligorijevicTT

Reply:
I am also fine with what @ajakovljevicTT suggested in the other comment, to create a test folder for ops which have very complex conversion patterns and to split their tests into multiple files, but I don't think that we truly had such a case so far. I think we could've just added those test cases into same file in different funcs of the module, but people opted otherwise based on their personal preferences.